crates/legacy/src/service.rs

```diff
@@ -1446,7 +1446,19 @@ impl<Types: NodeType> Iterator for HandleReceivedTxns<Types> {
         // encoded transaction length. Luckily, this being roughly proportional
         // to encoded length is enough, because we only use this value to estimate
         // our limitations on computing the VID in time.
-        let len = bincode::serialized_size(&tx).unwrap_or_default();
+        // In sequencer we account for block byte length limit with:
+        // let len = tx.payload().len() + NsTableBuilder::entry_byte_len() + NsPayloadBuilder::tx_table_header_byte_len() + NsPayloadBuilder::tx_table_entry_byte_len();
+        // where
+        // pub const fn entry_byte_len() -> usize {NS_ID_BYTE_LEN (4) + NS_OFFSET_BYTE_LEN (4)}
+        // pub const fn tx_table_header_byte_len() -> usize {NUM_TXS_BYTE_LEN (4)}
+        // pub const fn tx_table_entry_byte_len() -> usize {TX_OFFSET_BYTE_LEN (4)}
+        const NS_ID_BYTE_LEN: usize = 4;
+        const NS_OFFSET_BYTE_LEN: usize = 4;
+        const NUM_TXS_BYTE_LEN: usize = 4;
+        const TX_OFFSET_BYTE_LEN: usize = 4;
```

Comment:
These look good to me, but we may want to remove the duplicate definitions in the sequencer repo, so that the sequencer will import these values from the builder repo, to avoid inconsistency.